src/app/dataset-block/metadata-block/components/event-details/factory.events.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import { BasePropertyComponent } from "./components/common/base-property/base-property.component";
+import { MetadataEvent } from "./../../../../api/kamu.graphql.interface";
+import { SetPollingSource } from "src/app/api/kamu.graphql.interface";
+import { Type } from "@angular/core";
+import { SET_POLLING_SOURCE_DESCRIPTORS } from "./components/set-polling-source-event/set-polling-source-event.source";
+
+export interface EventRow {
+    label: string;
+    tooltip: string;
+    value?: unknown;
+    presentationComponent: Type<BasePropertyComponent>;
+}
+
+export interface EventSection {
+    title: string;
+    rows: EventRow[];
+}
+
+enum EventTypes {
+    SetPollingSource = "SetPollingSource",
+}
+
+enum SetPollingSourceProperty {
+    READ = "read",
+    FETCH = "fetch",
+    MERGE = "merge",
+    PREPROCESS = "preprocess",
+    PREPARE = "prepare",
+}
+
+abstract class EventRowFactory<TEvent> {
+    public abstract buildEventRows(event: TEvent): EventSection[];
+}
+
+class SetPollingSourceEventRowFactory extends EventRowFactory<SetPollingSource> {
+    public buildEventRows(event: SetPollingSource): EventSection[] {
+        const result: EventSection[] = [];
+        let rows: EventRow[] = [];
```

Comment:
`rows` is a local container of the section, it should not be defined at this level. It should be re-created in each section